pkg/git/git.go

```diff
@@ -53,14 +54,18 @@ func commitSingleFile(filePath, commitMessage, author string, worktree utilsWork
 }
 
 // PushChangesToRepository Pushes all committed changes in the repository to the remote repository
-func PushChangesToRepository(username, password string, force *bool, repository *git.Repository) error {
-	return pushChangesToRepository(username, password, force, repository)
+func PushChangesToRepository(username, password string, force *bool, repository *git.Repository, caCerts []byte) error {
+	return pushChangesToRepository(username, password, force, repository, caCerts)
 }
 
-func pushChangesToRepository(username, password string, force *bool, repository utilsRepository) error {
+func pushChangesToRepository(username, password string, force *bool, repository utilsRepository, caCerts []byte) error {
 	pushOptions := &git.PushOptions{
 		Auth: &http.BasicAuth{Username: username, Password: password},
 	}
+
+	if len(caCerts) > 0 {
+		pushOptions.CABundle = caCerts
+	}
```

Comment:
instead of checking len of `caCerts` we could directly add `caCerts` to `git.PushOptions`. If there are no links to custom certificates, `downloadCACertbunde` should return `nil`, so `CABundle` is nil (anyway its default value is nil), if there are certificates, `downloadCACertbunde` returns slice of bytes, then the slice will be added to `CABundle`.

Reply:
i added the check so that it does not effect the standard behaviour which i wanted to keep the same